src/mlpack/tests/string_encoding_test.cpp

```diff
@@ -501,6 +503,645 @@ BOOST_AUTO_TEST_CASE(CharExtractDictionaryEncodingSerialization)
+
+  encoder.Encode(stringEncodingInput, output, tokenizer);
+  const DictionaryType& dictionary = encoder.Dictionary();
+
+  // Checking that everything is mapped to different numbers
+  std::unordered_map<size_t, size_t> keysCount;
+  for (auto& keyValue : dictionary.Mapping())
+  {
+    keysCount[keyValue.second]++;
+    // Every token should be mapped only once
+    BOOST_REQUIRE_EQUAL(keysCount[keyValue.second], 1);
+  }
+  arma::mat expected = {
+    {  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 },
+    {  0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1,
+       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 },
+    {  1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 }
+  };
+  CheckMatrices(output, expected);
+}
+
+/**
+* Bag of Words encoding algorithm output saved in a vector.
+*/ 
+BOOST_AUTO_TEST_CASE(VectorBagOfWordsEncodingTest)
+{
+  using DictionaryType = StringEncodingDictionary<boost::string_view>;
+
+  vector<vector<size_t>> output;
+  BagOfWordsEncoding<SplitByAnyOf::TokenType> encoder(
+      (BagOfWordsEncodingPolicy()));
+  SplitByAnyOf tokenizer(" ,.");
+
+  encoder.Encode(stringEncodingInput, output, tokenizer);
+
+  const DictionaryType& dictionary = encoder.Dictionary();
+
+  // Checking that everything is mapped to different numbers
+  std::unordered_map<size_t, size_t> keysCount;
+  for (auto& keyValue : dictionary.Mapping())
+  {
+    keysCount[keyValue.second]++;
+    // Every token should be mapped only once
+    BOOST_REQUIRE_EQUAL(keysCount[keyValue.second], 1);
+  }
+
+  vector<vector<size_t>> expected = {
+    {  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 },
+    {  0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1,
+       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 },
+    {  1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 }
+  };
+
+  BOOST_REQUIRE(output == expected);
+}
+
+/**
+* Test Bag of Words encoding for characters.
+*/
+BOOST_AUTO_TEST_CASE(BagOfWordsEncodingIndividualCharactersTest)
+{
+  vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  BagOfWordsEncoding<CharExtract::TokenType> encoder;
+
+  // Passing a empty string to encode characters
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 1, 1, 1, 0, 0 },
+    { 0, 1, 1, 1, 1 },
+    { 1, 1, 0, 1, 0 }
+  };
+
+  CheckMatrices(output, target);
+}
+
+/**
+ * Test the Bag of Words encoding algorithm in case of individual
+ * character encoding, storing resulting in vector<vector<vectorType>>.
+ */
```

Comment:
```suggestion
  * Test the Bag of Words encoding algorithm in case of individual
  * character encoding, storing resulting in vector<vector<vectorType>>.
  */
```

Reply:
Done